### DISCUSSION TITLE
Are practice results no longer available?

### DISCUSSION THREAD

**Discussion by oneillcode** [theOehrly/Fast-F1#391] [General]
Hello,

I used to use this API to gather practice results after the races but it seems that is no longer possible. I have tried using all the different 'backend' options and they all provide nan values in Python. Here is an example:

`{'DriverNumber': '1', 'BroadcastName': 'M VERSTAPPEN', 'Abbreviation': 'VER', 'DriverId': 'max_verstappen', 'TeamName': 'Red Bull Racing', 'TeamColor': '3671C6', 'TeamId': 'red_bull', 'FirstName': 'Max', 'LastName': 'Verstappen', 'FullName': 'Max Verstappen', 'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/M/MAXVER01_Max_Verstappen/maxver01.png.transform/1col/image.png', 'CountryCode': 'NED', 'Position': nan, 'ClassifiedPosition': '', 'GridPosition': nan, 'Q1': NaT, 'Q2': NaT, 'Q3': NaT, 'Time': NaT, 'Status': '', 'Points': nan}`

I was able to capture results for Monaco and earlier but now those races also do not provide results.

Thanks!

**Comment by theOehrly** [comment]
This is likely related to #384 but that should actually be mostly resolved now.

I'm not sure what you mean with "practice results", though? Results for FP sessions?

**Comment by oneillcode** [comment]
@theOehrly Yes FP1, FP2 and FP3.

The code I have been using is:

`
            session = fastf1.get_session(year, round, 'FP1')

            session.load(telemetry=False, laps=True, weather=True)

            session_dict = session.results.to_dict('records')
`

It does look like `session.laps` returns all the lap times with some NaT mixed in. I can try and look at the code to see if something is tripping it up to provide NaT now.

**Reply by theOehrly** [reply]
I still don't quite understand.

> {'DriverNumber': '1', 'BroadcastName': 'M VERSTAPPEN', 'Abbreviation': 'VER', 'DriverId': 'max_verstappen', 'TeamName': 'Red Bull Racing', 'TeamColor': '3671C6', 'TeamId': 'red_bull', 'FirstName': 'Max', 'LastName': 'Verstappen', 'FullName': 'Max Verstappen', 'HeadshotUrl': 'https://www.formula1.com/content/dam/fom-website/drivers/M/MAXVER01_Max_Verstappen/maxver01.png.transform/1col/image.png', 'CountryCode': 'NED', 'Position': nan, 'ClassifiedPosition': '', 'GridPosition': nan, 'Q1': NaT, 'Q2': NaT, 'Q3': NaT, 'Time': NaT, 'Status': '', 'Points': nan}

Which of these values should not be ``nan`` in a practice session? I think none of those values that are missing are applicable to practice sessions?

**Reply by oneillcode** [reply]
@theOehrly 

Sorry after looking at my code, the only thing that seems to have changed is that position is now 'nan' for every driver. I thought that it was also providing the lap time under 'Time' but I calculate that myself.

I can just start to calculate the position myself so its not a big deal. Sorry about that and thanks a ton for maintaining this API!

**Reply by theOehrly** [reply]
Ok, but now I have something concrete to investigate. I can check if a change in FastF1 is responsible for this.
